src/PokemonsList.tsx

```diff
@@ -0,0 +1,32 @@
+import PokemonCard from './PokemonCard';
+import useIntersectionObserver from './useIntersectionObserver';
+import { useState, useRef, useEffect } from 'react';
+
+const pageSize = 2;
+
+export default function PokemonsList(): JSX.Element {
+  const [count, setCount] = useState(0);
+  const ref = useRef(null);
+  const isBottomVisible: unknown = useIntersectionObserver(ref, {
+    threshold: 1,
+  });
+
+  useEffect(() => {
+    if (isBottomVisible) setCount(count + 1);
+  }, [isBottomVisible]);
```

Comment:
Most likely it will not work on big screens because it should be based on events, not on value. This hook will not notice if the value is set to `true` few times in a raw.